tools/run_local_ironic.sh

```diff
@@ -170,9 +170,20 @@ fi
 
 BASIC_AUTH_MOUNTS=""
 IRONIC_HTPASSWD=""
+IRONIC_HTPASSWD_FILE="/etc/ironic/htpasswd"
 if [ -n "$IRONIC_USERNAME" ]; then
-     IRONIC_HTPASSWD="$(htpasswd -n -b -B "${IRONIC_USERNAME}" "${IRONIC_PASSWORD}")"
-     IRONIC_HTPASSWD="--env HTTP_BASIC_HTPASSWD=${IRONIC_HTPASSWD} --env IRONIC_HTPASSWD=${IRONIC_HTPASSWD}"
+     HTPASSWD="$(htpasswd -n -b -B "${IRONIC_USERNAME}" "${IRONIC_PASSWORD}")"
+     HTPASSWD=$(echo -n "${HTPASSWD}"|base64 -w 0)
+     echo "apiVersion: v1
+data:
+     htpasswd: ${HTPASSWD}
+kind: Secret
+metadata:
+     name: ironic-htpasswd
+     namespace: ${NAMEPREFIX}-system
+type: Opaque" >> "${IRONIC_HTPASSWD_FILE}"
+
+     IRONIC_HTPASSWD="-v ${IRONIC_HTPASSWD_FILE}:${IRONIC_HTPASSWD_FILE}"
```

Comment:
Secret are not mounted this way in Kubernetes. It works locally, sure, but I don't think you can make Kubernetes even do it without a lot of magic. See my comments on ironic-image. Here you'll basically mount 2 files: with username and password.